### ISSUE TITLE
Mac OS darwin doesn't use .exe once make install is run

### ISSUE THREAD

**Issue by tonybaloney** [python/performance#37]
During the compile_all stage, the script tries to print the version on the screen.

If install=True in the configuration, `make install` is run. Darwin does compile the CPython binary as `python.exe` but once you run `make install` it corrects it to the right name.

I can see the fault in the logic here.
https://github.com/python/performance/blob/f9ebee09652f77ca01736f9b034a0bb0d8826a45/performance/compile.py#L301

```
2018-03-31 20:20:25,352: Installed Python version:
2018-03-31 20:20:25,353: + /Users/anthonyshaw/repo/python_comparison/bench_tmpdir/prefix/bin/python3.exe --version
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/__main__.py", line 2, in <module>
    performance.cli.main()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/cli.py", line 226, in main
    _main()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/cli.py", line 190, in _main
    cmd_compile(options)
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 921, in cmd_compile
    bench.main()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 656, in main
    failed = self.compile_bench()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 641, in compile_bench
    self.compile_install()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 459, in compile_install
    self.python.compile_install()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 372, in compile_install
    self.get_version()
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 312, in get_version
    self.run(self.program, '--version')
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 63, in run
    self.app.run(*cmd, cwd=self.cwd, **kw)
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 195, in run
    exitcode = self.run_nocheck(*cmd, **kw)
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 174, in run_nocheck
    proc = self.create_subprocess(cmd, **kwargs)
  File "/Users/anthonyshaw/repo/python_comparison/env/lib/python3.6/site-packages/performance/compile.py", line 158, in create_subprocess
    return subprocess.Popen(cmd, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 707, in __init__
    restore_signals, start_new_session)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 1326, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/anthonyshaw/repo/python_comparison/bench_tmpdir/prefix/bin/python3.exe'
2018-03-31 20:20:25,405: Command /Users/anthonyshaw/repo/python_comparison/env/bin/python3.6 -m performance compile /Users/anthonyshaw/repo/python_comparison/testing.conf 3.6 3.6 --no-update --no-tune failed with exit code 1
```

**Comment by vstinner** [comment]
Do you want to work on a fix? (a pull request)

**Comment by tonybaloney** [comment]
Sure. I had a fix in my own fork anyway, can merge it back

**Comment by vstinner** [comment]
Fixed by commit 2a9524c0a5714e85106671bc61d750e800fe17db if I understood correctly.
